src/novel_applications/defi/clob.md

```diff
@@ -0,0 +1,167 @@
+# Private Order Book
+
+<!-- add header image -->
+
+## Order Books: A Powerful Financial Product
+
+I want to discuss a PoC of an on-chain order book exchange on Polygon Miden. Miden is perfectly suitable for even a private on-chain order book exchange leveraging client-side proving, high throughput, updateable transactions, and the security of Ethereum. 
+
+An order book refers to an electronic list of buy and sell orders for a specific security or financial instrument organized by price level. There are three parts to an order book: buy orders, sell orders, and order history. An order book is dynamic and constantly updated in real-time throughout the day. A key feature of order book models is allowing users to submit two types of orders: market or limit orders. Market orders mean a trader instantly buys or sells at the best price (highest bid, lowest ask). Limit orders are not instant and buy or sell at a specific price. Next to the order book, there also must be an order matching system following some rules of best execution to have an order book exchange. 
+
+In crypto markets, the dominant venue for exchange has been centralized order book-based exchanges (CEX). There is high demand for on-chain order-book exchanges; see [here](https://twitter.com/EliBenSasson/status/1614575237122629633?s=20) 40 TPS only on dYdX. Onchain exchanges are mostly AMM-based exchanges that cannot provide quite the same benefits: 
+
+Order-book exchanges:
+- Provide price efficiency
+- Reduce the risk of slippage and have no impermanent loss
+- Are widely accepted by both institutional and retail traders alike 
+- Are easier to use
+
+To run an on-chain order book, several features are needed:
+
+Needed:
+1. High TPS - builders ask for 30-40 and some even more
+2. Low finality - sub-second finality would be optimal
+3. Quick and cheap order updates - Order updates should be almost instantaneous and free, especially for Market Makers
+
+Nice to have:
+4. Privacy at the base layer - to not reveal a trading strategy 
+5. Matching engine - FIFO/Pro-Rata should also run on-chain to ensure "best execution"
+6. MEV protection - to ensure "best execution"
+
+(Are there more required features? Do we need to cancel the order quickly?)
```

Comment:
I think you can delete this 

(Are there more required features? Do we need to cancel the order quickly?)